### PR TITLE
feat(packaging): move packaging to a single "@angular/material"

### PR DESCRIPTION
**Pull request by hansl**
Moving forward, we will only publish a single `@angular/material` package for the Angular2 Material project. This simplifies installation and usage for everyone, and won't increase the size of the library for people using AoT. It is also assumed that most people who use Angular Material will use most of the components so using a single UMD bundle should not increase their package size too much in JIT.

# BREAKING CHANGES

People using `npm install @angular2-material/COMPONENT_NAME` need to remove all their `@angular2-material` dependencies and use a single `@angular/material` dependency instead.

### CODE REVIEW

File: tools/gulp/tasks/components.ts [hunk 1]
```diff
@@ -63,48 +53,38 @@ task(':build:components:rollup', [':build:components:ts'], () => {
...
-        const result = bundle.generate({
-          moduleName: `md.${camelCase(name)}`,
-          format: 'umd',
-          globals
-        });
-        const outputPath = path.join(DIST_COMPONENTS_ROOT, name, `${name}.umd.js`);
-        writeFileSync( outputPath, result.code );
-      });
-  }, Promise.resolve());
+  // Build the all package.
```
Comment (jelbourn) [inline tools/gulp/tasks/components.ts]: Comment something like "Use rollup to create the published umd bundle." ?

**Comment by kyleraymorgan** [comment]
When should we start using `npm install @angular/material`?

File: src/demo-app/system-config.ts [hunk 2]
```diff
@@ -78,27 +25,40 @@ const barrels: string[] = [
   'live-announcer',
   'portal',
   'overlay',
-  ...components
-  /** @cli-barrel */
 ];
 
-const _cliSystemConfig: any = angularPackages;
+
+const angularPackages: any = angularDeps.reduce((config: any, pkg: string) => {
```
Comment (hansl) [inline src/demo-app/system-config.ts]: System-Config has been simplified. PTAL

File: src/demo-app/system-config.ts [hunk 3]
```diff
@@ -78,27 +25,40 @@ const barrels: string[] = [
...
+  config[pkg] = { main: `bundles/${pkg.replace('@angular/', '')}.umd.js` };
+  return config;
+}, {});
 barrels.forEach((barrelName: string) => {
-  _cliSystemConfig[barrelName] = { main: 'index' };
+  angularPackages[barrelName] = { main: 'index' };
 });
 
+const angularMaps: any = angularDeps.reduce((config: any, pkg: string) => {
```
Comment (hansl) [inline src/demo-app/system-config.ts]: System-Config has been simplified. PTAL

File: tools/gulp/tasks/components.ts (hunk 1 above)
Comment (hansl) [inline tools/gulp/tasks/components.ts]: Done.

**Comment by hansl** [comment]
@jelbourn: this is ready to be reviewed.

Sourcemaps are still messy (they give the `tsc` output), but for now this will unblock you for AC. I'll improve in a subsequent PR.

File: src/lib/core/theming/_all-theme.scss [hunk 4]
```diff
@@ -1,23 +1,23 @@
...
+@import '../../menu/menu-theme';
+@import '../../progress-bar/progress-bar-theme';
+@import '../../progress-circle/progress-circle-theme';
+@import '../../radio/radio-theme';
+@import '../../sidenav/sidenav-theme';
+@import '../../slide-toggle/slide-toggle-theme';
+@import '../../slider/slider-theme';
+@import '../../tabs/tabs-theme';
+@import '../../toolbar/toolbar-theme';
+@import '../../tooltip/tooltip-theme';
```
Comment (jelbourn) [inline src/lib/core/theming/_all-theme.scss]: Want to move this mixin into `_theming.scss` and just delete this file?

**Comment by hansl** [comment]
Okay. Will do that when I'm back from dinner

File: src/lib/core/theming/_all-theme.scss (hunk 4 above)
Comment (hansl) [inline src/lib/core/theming/_all-theme.scss]: Done.

File: src/lib/core/theming/_theming.scss [hunk 5]
```diff
@@ -82,3 +82,51 @@
...
+@import '../../sidenav/sidenav-theme';
+@import '../../slide-toggle/slide-toggle-theme';
+@import '../../slider/slider-theme';
+@import '../../tabs/tabs-theme';
+@import '../../toolbar/toolbar-theme';
+@import '../../tooltip/tooltip-theme';
+
+
+// Create a theme.
+@mixin angular-material-theme($theme) {
```
Comment (jelbourn) [inline src/lib/core/theming/_theming.scss]: As discussed: internal presubmit revealed that moving this here caused an unresolvable circular dependency. Need to move it back to it's own file. I suggest we call it `_all-component-themes.scss`

Comment (hansl) [inline src/lib/core/theming/_theming.scss]: Dono

**Pull request merged**